server/api/migrations/versions/67b0169ba6b4_add_alert_history.py

```diff
@@ -0,0 +1,107 @@
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "alert_history",
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column(
+            "activation_time", mysql.DATETIME(timezone=True, fsp=3), nullable=False
+        ),
+        sa.Column(
+            "name", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "project", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column("data", sa.JSON(), nullable=True),
+        sa.Column(
+            "entity_id", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "entity_kind",
+            sa.String(length=255, collation="utf8mb3_bin"),
+            nullable=False,
+        ),
+        sa.Column(
+            "event_kind", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column(
+            "severity", sa.String(length=255, collation="utf8mb3_bin"), nullable=False
+        ),
+        sa.Column("number_of_events", sa.Integer(), nullable=False),
+        sa.PrimaryKeyConstraint("activation_time", "id", name="_alert_history_uc"),
+    )
+    op.create_index(
+        "ix_alert_history_entity_id", "alert_history", ["entity_id"], unique=False
+    )
+    op.create_index(
+        "ix_alert_history_entity_kind", "alert_history", ["entity_kind"], unique=False
+    )
+    op.create_index(
+        "ix_alert_history_event_kind", "alert_history", ["event_kind"], unique=False
+    )
+    op.create_index("ix_alert_history_name", "alert_history", ["name"], unique=False)
+    op.create_index(
+        "ix_alert_history_project", "alert_history", ["project"], unique=False
+    )
+    op.create_index(
+        "ix_alert_history_severity", "alert_history", ["severity"], unique=False
+    )
+    # ### end Alembic commands ###
+
+    # Add initial partitioning
+    partition_interval = os.getenv("PARTITION_INTERVAL", "YEARWEEK")
+    default_partition_value = os.getenv("DEFAULT_PARTITION_VALUE", "2023-01-01")
+    partition_sql = f"""
+         ALTER TABLE alert_history
+         PARTITION BY RANGE ({partition_interval}(activation_time)) (
+             PARTITION p0 VALUES LESS THAN ({partition_interval}('{default_partition_value}'))
+         );
+     """
+    op.execute(partition_sql)
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_index("ix_alert_history_severity", table_name="alert_history")
+    op.drop_index("ix_alert_history_project", table_name="alert_history")
+    op.drop_index("ix_alert_history_name", table_name="alert_history")
+    op.drop_index("ix_alert_history_event_kind", table_name="alert_history")
+    op.drop_index("ix_alert_history_entity_kind", table_name="alert_history")
+    op.drop_index("ix_alert_history_entity_id", table_name="alert_history")
```

Comment:
Do you need to drop the indexes before dropping the table? I think this is redundant.